### ISSUE TITLE
VTK output file crashes paraview (paraview version 5.7.0)

### ISSUE THREAD

**Issue by firstkingofrome** [nschloe/meshio#643]
Hello I have a number of gmesh files that I would like to convert to vtk. The vtk file I produce (I tried both ascii and binary) cause paraview to crash when I attempt to render them. I have attached an input gmesh file and output ascii vtk. file.
I converted by simply loading a test file and attempting to rewrite it as a vtk file:
import meshio
testFile = "gmshRes.mesh.0.msh"
msh = meshio.read(testFile) 
meshio.write("test.vtk",msh,file_format="vtk-ascii")

Based on this discussion https://discourse.paraview.org/t/paraview-immediately-crashing-upon-when-opening-a-legacy-vtk-file-created-using-python/1104 I thought that it might be a problem with the CELLS block?

Thank You for helping me look into this issue!

(Note I changed added a .txt extension to the gmesh and vtk files I was describing in order to get github to accept them)

[gmshRes.mesh.0.msh.txt](https://github.com/nschloe/meshio/files/4100637/gmshRes.mesh.0.msh.txt)

[test.vtk.txt](https://github.com/nschloe/meshio/files/4100643/test.vtk.txt)

**Comment by gdmcbain** [comment]
That test.vtk crashes Gmsh 4.5.1 too.

I note that MSH 2.2 can also be converted to VTK by Gmsh with
```shell
gmsh -2 gmshRes.mesh.0.msh -o gmshRes.mesh.0.vtk
```
which gives [gmshRes.mesh.0.vtk.txt](https://github.com/nschloe/meshio/files/4100727/gmshRes.mesh.0.vtk.txt) which doesn't look too bad (opens in ParaView 5.6.0, e.g.).

This might be helpful for seeing where meshio is going wrong here.  I see that the original MSH 2.2 has 3784 nodes as does the meshio VTK but the Gmsh VTK has only 3272.

**Comment by gdmcbain** [comment]
Gmsh 4.5.1 crashes pretty quickly on opening test.vtk, but I caught the error messages in 
[test.txt](https://github.com/nschloe/meshio/files/4100759/test.txt) with
```shell
gmsh -log test.txt test.vtk 
```
Looks like:
```
Info: Reading 'test.vtk'...
Info: Reading 3784 points
Info: Reading 2564 cells
Error: Wrong node index 6711
Error: Wrong node index 6710
…
```
and many more like the last two.

I see that in the original MSH 2.2, the nodes aren't numbered contiguously from 1 but begin at 17 and increase noncontiguously to 16131.  That might be confusing meshio.

**Comment by gdmcbain** [comment]
The issue isn't related to writing VTK at all; it's on reading MSH 2.2 with sparsely numbered nodes  Just doing:
```python
mesh = meshio.read('gmshRes.mesh.0.msh')
```
we can see:
```
>>> mesh.cells[0].data.min(0)
array([  17,  776,  758,   16,   48, 1051, 1051,   48])
>>> mesh.cells[0].data.max(0)
array([14355, 14354, 14354, 14355, 14757, 14756, 14756, 14757])
```
whereas
```
>>> mesh.points.shape
(3784, 3)
```
i.e. the connectivity arrays in mesh.cells are trying to index outside mesh.points.

I thought meshio handled sparsely numbered nodes in MSH 2.2.  Is this a regression?

**Comment by nschloe** [comment]
To be honest, I have zero motivation for getting into msh 2.2. @firstkingofrome: My advice: Update to a newer gmsh version that uses mesh format 4.1, there's a good chance this is giving you less trouble. I'll  also be happy to review PRs of course.

**Comment by firstkingofrome** [comment]
Thank You for letting me know! Ill probably try producing the meshes using version 4.1 of the format as you suggest.

**Comment by nschloe** [comment]
Fixed via #668.
